File: SDKs/googleapis.com/recommendationengine/v1beta1/python/src/sdk/models/shared/googlecloudrecommendationenginev1beta1usereventinlinesource.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class GoogleCloudRecommendationengineV1beta1UserEventInlineSource:
    r"""GoogleCloudRecommendationengineV1beta1UserEventInlineSource
    The inline source for the input config for ImportUserEvents method.
    """
    
    user_events: Optional[List[GoogleCloudRecommendationengineV1beta1UserEvent]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('userEvents') }})
    

File: SDKs/googleapis.com/recommendationengine/v1beta1/typescript/src/sdk/models/shared/googlecloudrecommendationenginev1beta1usereventinlinesource.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { GoogleCloudRecommendationengineV1beta1UserEvent } from "./googlecloudrecommendationenginev1beta1userevent";



// GoogleCloudRecommendationengineV1beta1UserEventInlineSource
/** 
 * The inline source for the input config for ImportUserEvents method.
**/
export class GoogleCloudRecommendationengineV1beta1UserEventInlineSource extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=userEvents", elemType: GoogleCloudRecommendationengineV1beta1UserEvent })
  userEvents?: GoogleCloudRecommendationengineV1beta1UserEvent[];
}
